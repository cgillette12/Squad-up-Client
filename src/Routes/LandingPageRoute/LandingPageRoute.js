import React from 'react'
import LandingPage from '../../components/Landing-Page/Landing-Page'

export default function LandingPageRoute() {
  return (
    <LandingPage />
  )
}
