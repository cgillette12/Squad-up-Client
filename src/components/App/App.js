import React from 'react'
import { Route, Switch } from 'react-router-dom'
import Header from '../Header/Header'
import LoginRoute from '../../Routes/LoginRoute/LoginRoute'
import LandingPageRoute from '../../Routes/LandingPageRoute/LandingPageRoute'
import RegisterRoute from '../../Routes/RegisterRoute/RegisterRoute'
import NewSquadRoute from '../../Routes/NewSquadRoute/NewSquadRoute'
import DashboardRoute from '../../Routes/DashboardRoute/DashboardRoute'
import './App.css'
import LiveChat from '../LiveChat/LiveChat'

function App() {
  return (
    <div className="App">
      <Header />
      <main role="main">
       <Switch>
         <Route exact path={'/'} component={LandingPageRoute} />
         <Route exact path={'/login'} component={LoginRoute} />
         <Route exact path={'/register'} component={RegisterRoute} />
         <Route exact path={'/newsquad'} component={NewSquadRoute}/>
         <Route path={'/dashboard'} component={DashboardRoute} />
       </Switch>
      </main>
    </div>
  )
}

export default App
