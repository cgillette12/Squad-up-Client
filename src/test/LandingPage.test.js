import React from 'react';
import { expect } from 'chai';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../components/App/App'
import { MemoryRouter } from 'react-router-dom'
import LandingPageRoute from '../Routes/LandingPageRoute/LandingPageRoute';
Enzyme.configure({ adapter: new Adapter() });

it('renders the landing page without crashing and renders the correct text', () => {
    const wrapper = mount(
        <MemoryRouter initialEntries={['/']}>
            <App/>
        </MemoryRouter>
    )
    expect(wrapper.find(LandingPageRoute)).to.have.length(1)
    expect(wrapper.text()).equal('Squad UpRegisterLoginHomeSquadsChatLoginLogoutWelcome to Squad Up!Squad Up, allows its users to look through many games and form a community. Feel free to register and have a look around!RegisterLogin')
})
it('renders both register and log in button', () => {
    const wrapper = mount(
        <MemoryRouter initialEntries={['/']}>
            <App/>
        </MemoryRouter>
    )
    expect(wrapper.find('.link-button')).to.have.length(2)
    expect(wrapper.find('.link-button').at(0).text()).equal('Register')
    expect(wrapper.find('.link-button').at(1).text()).equal('Login')
})

