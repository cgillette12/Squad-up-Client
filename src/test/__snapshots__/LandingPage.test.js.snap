// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the LandingPage as expected 1`] = `
<div
  id="homePage"
>
  <h1
    className="landing-page-heading"
  >
    SquadUp
  </h1>
  <h2
    className="tag-line"
  >
    One platform. Any game.
  </h2>
  <div
    id="features-wrapper"
  >
    <img
      alt="Dashboard Screenshot"
      className="image"
      src="DashBoardWithUpdatedChat.png"
    />
    <div
      id="content-wrapper"
    >
      <p
        className="image-caption"
      >
        Everyone loves games. But finding a community for your specific game can leave you with a bunch of accounts and apps to download. Not anymore. SquadUp is the new home for all of your gaming interests. From the latest FPS and MOBAs, to the oldest table-top RPGs and even IRL sports, we've got it all.
        <br />
        <br />
        The SquadUp dashboard is the last tool you'll need to connect with fellow gamers. Search for games, join or create squads and stay connected to your squadmates with a dedicated squad chatroom. 
        <br />
        <br />
        Check us out before signing up!
      </p>
      <div
        id="links-wrapper"
      >
        <a
          href="/Register"
          onClick={[Function]}
        >
          <button
            className="link-button"
          >
            Sign up
          </button>
        </a>
        <a
          href="/Login"
          onClick={[Function]}
        >
          <button
            className="link-button"
          >
            Login
          </button>
        </a>
        <button
          className="link-button"
          onClick={[Function]}
        >
          Demo
        </button>
      </div>
    </div>
  </div>
</div>
`;
